test/functional/wallet_avoidreuse.py

```diff
@@ -240,5 +242,26 @@ def test_fund_send_fund_send(self):
         assert_approx(self.nodes[1].getbalance(), 1, 0.001)
         assert_approx(self.nodes[1].getbalance(avoid_reuse=False), 11, 0.001)
 
+    def test_destination_groupings(self):
+        '''
+        If a destination has more than 10 outputs, exactly 10 outputs should
```

Comment:
what assumptions does this test have?

Seems like it assumes the wallet has no more than 10*21 BTC? In other words, those sends completely drain the wallet of other UTXO addresses. This should be asserted and/or programmatically done.

Reply:
It took me awhile to realize what is being stated here.

If a destination has more than 10 outputs, the 10-output groups should always be picked before the <10-output group, if any such exists.

I.e. if a destination has 19 outputs, it will result in one 10-output group and one 9-output group. The second should never ever be used unless the former is also used.

Reply:
I have expanded the test and the comments and hope it's more clear now.